virtcontainers/container.go

```diff
@@ -652,7 +652,7 @@ func (c *Container) stop() error {
 	// return an error, but instead try to kill it forcefully.
 	if err := waitForShim(c.process.Pid); err != nil {
 		// Force the container to be killed.
-		if err := c.sandbox.agent.killContainer(c.sandbox, *c, syscall.SIGKILL, true); err != nil {
+		if err := c.kill(syscall.SIGKILL, true); err != nil {
```

Comment:
Why not just call into c.signalProcess here. `c.kill` api is not really required.

Reply:
It's a helper function for cases like this when we want to send signals to the container process itself. And it hides the detail that we use `c.process.Token` as process ID of the container process, useful because it's also called by `KillContainer()` in api.go.